lib/Dialect/AIEVec/Transforms/ConvertVectorToAIEVec.cpp

```diff
@@ -1119,6 +1119,121 @@ using LowerVectorMinFOpToAIEVecMinOp =
+  case CmpFPredicate::OGT:
+    return CmpIPredicate::ugt;
+  case CmpFPredicate::UGE:
+  case CmpFPredicate::OGE:
+    return CmpIPredicate::uge;
+  case CmpFPredicate::ULT:
+  case CmpFPredicate::OLT:
+    return CmpIPredicate::ult;
+  case CmpFPredicate::ULE:
+  case CmpFPredicate::OLE:
+    return CmpIPredicate::ule;
+  case CmpFPredicate::UNE:
+  case CmpFPredicate::ONE:
+    return CmpIPredicate::ne;
+  default:
+    llvm_unreachable("Unexpected predicate!");
+  }
+}
+
+static arith::CmpIPredicate
+convertToIntegerPredicate(arith::CmpIPredicate pred) {
+  return pred;
+}
+
+static aievec::CmpOp createCmpOpAieML(ConversionPatternRewriter &rewriter,
+                                      CmpIPredicate pred, Location loc,
+                                      Type type, Value lhs, Value rhs) {
+  switch (pred) {
+  case CmpIPredicate::eq:
+    return rewriter.create<aievec::CmpOp>(loc, type, lhs, rhs, "eq");
+  case CmpIPredicate::ne:
+    return rewriter.create<aievec::CmpOp>(loc, type, lhs, rhs, "ne");
+  case CmpIPredicate::slt:
+  case CmpIPredicate::ult:
+    return rewriter.create<aievec::CmpOp>(loc, type, lhs, rhs, "lt");
+  case CmpIPredicate::sle:
+  case CmpIPredicate::ule:
+    return rewriter.create<aievec::CmpOp>(loc, type, lhs, rhs, "le");
+  case CmpIPredicate::sgt:
+  case CmpIPredicate::ugt:
+    return rewriter.create<aievec::CmpOp>(loc, type, lhs, rhs, "gt");
+  case CmpIPredicate::sge:
+  case CmpIPredicate::uge:
+    return rewriter.create<aievec::CmpOp>(loc, type, lhs, rhs, "ge");
+  }
+  return nullptr;
+}
+
+template <typename SrcOpTy, typename CmpTy>
+struct LowerVectorCmpOpToAIEVecCmpOp : public OpConversionPattern<SrcOpTy> {
+  using OpConversionPattern<SrcOpTy>::OpConversionPattern;
+  using OpAdaptor = typename SrcOpTy::Adaptor;
+
+  LowerVectorCmpOpToAIEVecCmpOp(MLIRContext *context)
+      : OpConversionPattern<SrcOpTy>(context) {}
+
+  LogicalResult
+  matchAndRewrite(SrcOpTy srcOp, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    VectorType lhsType = dyn_cast<VectorType>(srcOp.getLhs().getType());
+    if (!lhsType)
+      return failure();
+
+    llvm::SmallSet<unsigned, 16> elWidthSet;
+    elWidthSet.insert(8);
+    elWidthSet.insert(16);
+    elWidthSet.insert(32);
+
+    Type scalarType = lhsType.getElementType();
+    unsigned elWidth = scalarType.getIntOrFloatBitWidth();
+    unsigned laneSize = getVectorLaneSize(lhsType);
+
+    if (!(elWidthSet.count(elWidth) && laneSize * elWidth == 512))
+      return failure();
+
+    // Unsigned int and unsigned long long are acceptable type.
+    Type type =
+        mlir::IntegerType::get(srcOp.getContext(), laneSize <= 32 ? 32 : 64,
+                               mlir::IntegerType::Unsigned);
+
+    Location loc = srcOp.getLoc();
+    Value lhs = srcOp.getLhs();
+    Value rhs = srcOp.getRhs();
+    CmpTy pred = srcOp.getPredicate();
+
+    arith::CmpIPredicate ipred = convertToIntegerPredicate(pred);
```

Comment:
Do you allow the pattern to apply if the LHS is signed and the RHS is unsigned?

Reply:
According to aie-ml intrinsics, we do not handle the comparison between unsigned and signed. In AIEVecOps.td, I also request for AllTypesMatch for lhs and rhs.